doc/release-notes-0.4.0.rst

```diff
@@ -0,0 +1,274 @@
+
+Documentation
+=============
+
+* Added or updated documentation for the following samples:
+
+  * :ref:`asset_tracker`, :ref:`nrf_coap_client_sample`,  :ref:`mqtt_simple_sample`, and :ref:`http_application_update_sample`
+  * :ref:`peripheral_hids_mouse`, :ref:`peripheral_hids_keyboard`,  :ref:`bluetooth_central_hids`, and :ref:`bluetooth_central_dfu_smp`
+  * :ref:`record_text`, :ref:`writable_ndef_msg`, and :ref:`nfc_tag_reader`
+  * :ref:`event_manager_sample` and :ref:`profiler_sample`
+
+* Added or updated documentation for the following libraries:
+
+  * :ref:`bt_conn_ctx_readme`
+  * :ref:`dk_buttons_and_leds_readme`
+  * :ref:`nfc`
+  * :ref:`event_manager` and :ref:`profiler`
+  * :ref:`at_cmd_parser_readme`, :ref:`at_params_readme`, and :ref:`modem_info_readme`
+  * :ref:`lib_download_client`
+  * :ref:`lib_spm`
+  * :ref:`st25r3911b_nfc_readme`
+  * :ref:`dfu_smp_c_readme` and :ref:`lbs_readme`
+
+* Added or updated the following documentation:
+
+  * :ref:`gs_installing` - updated for west, added instructions for macOS
+  * :ref:`gs_programming` - updated for new SES version
+  * :ref:`gs_modifying` - added
+  * :ref:`ug_unity_testing` - added
+  * :ref:`ug_pm` - added
+  * :ref:`ug_nrf9160` - updated Secure Partition Manager and Network mode sections
+
+
+Known issues
+************
+
+nRF9160
+=======
+
+* The :ref:`asset_tracker` sample does not wait for connection to nRF Cloud before trying to send data.
+  This causes the sample to crash if the user toggles one of the switches before the board is connected to the cloud.
+* The :ref:`asset_tracker` sample might show up to 2.5 mA current consumption in idle mode with ``CONFIG_POWER_OPTIMIZATION_ENABLE=y``.
+* If a debugger (for example, J-Link) is connected via SWD to the nRF9160, the modem firmware will reset.
+  Therefore, the LTE modem cannot be operational during debug sessions.
+* The SEGGER Control Block cannot be found by automatic search by the RTT Viewer/Logger.
+  As a workaround, set the RTT Control Block address to 0 and it will try to search from address 0 and upwards.
+  If this does not work, look in the ``builddir/zephyr/zephyr.map`` file to find the address of the ``_SEGGER_RTT`` symbol in the map file and use that as input to the viewer/logger.
+
+Crypto
+======
+
+* The :ref:`nrfxlib:nrf_security` glue layer is broken because symbol renaming is not handled correctly.
+  Therefore, the behavior is undefined when selecting multiple back-ends for the same algorithm (for example, AES).
+
+
+Subsystems
+==========
+
+Bluetooth Low Energy
+--------------------
+
+* :ref:`peripheral_lbs` does not report the Button 1 state correctly.
+* The central samples (:ref:`central_uart`, :ref:`bluetooth_central_hids`) do not support any pairing methods with MITM protection.
+* The :ref:`gatt_dm_readme` is not supported on nRF51 devices.
+* Bluetooth LE samples cannot be built with the :ref:`nrfxlib:ble_controller` v0.1.0.
```

Comment:
Correct?

Reply:
yes

Reply:
I think it's correct. _BTW one could ask why do we provide this BLE Controller at all in 0.4.0_?